hoomd/mpcd/DryingDropletStreamingMethod.h

```diff
@@ -0,0 +1,311 @@
+        mpcd::detail::boundary m_bc;                       //!< Boundary condition
+        Scalar m_density;                                  //!< Solvent density
+        unsigned int m_seed;                               //!< Seed to evaporator pseudo-random number generator
+        unsigned int m_Npick;                              //!< Number of particles picked for evaporation on this rank
+        const unsigned int m_mask = 1 << 1;                //!< Mask for flags
+
+        GPUArray<unsigned int> m_bounced_index;            //!< Indices of bounced particles
+        GPUVector<unsigned int> m_picks;                   //!< Particles picked for evaporation on this rank
+        GPUVector<mpcd::detail::pdata_element> m_removed;  //!< Hold output particles that are removed
+
+        virtual void applyPicks();                         //!< Apply the picks
+
+    private:
+        std::vector<unsigned int> m_all_picks;             //!< All picked particles on all the ranks
+
+        unsigned int calculateNumbounced();                //!< For calculating N_bounced and m_bounced_index 
+        //!< For Making a random pick of particles across all ranks
+        void makeAllPicks(unsigned int timestep, unsigned int N_pick, unsigned int N_bounced_total);
+    };
+
+/*!
+ * \param timestep Current time to stream
+ */
+void DryingDropletStreamingMethod::stream(unsigned int timestep)
+    {
+    // use peekStream since shouldStream will be called by parent class 
+    if(!peekStream(timestep)) return;
+    
+    // compute final Radius and Velocity of surface
+    const Scalar start_R = m_R->getValue(timestep);
+    const Scalar end_R = m_R->getValue(timestep + m_period);
+    const Scalar V = (end_R - start_R)/(m_mpcd_dt);
+    // checks if V <= 0, since size of droplet must decrease
+    if (V > 0)
+        {
+        throw std::runtime_error("Droplet radius must decrease.");
+        }
+
+    /*
+     * If the initial geometry was not set. Set the geometry and validate the geometry 
+     * Because the interface is shrinking, it is sufficient to validate only the first time the geometry
+     * is set.
+     */
+    if (!m_geom)
+        {
+        m_geom = std::make_shared<mpcd::detail::SphereGeometry>(start_R, V , m_bc ); 
+        validate();
+        m_validate_geom = false;
+        }
+
+    /*
+     *Update the geometry radius to the size at the end of the streaming step.
+     * This needs to be done every time.
+     */
+    m_geom = std::make_shared<mpcd::detail::SphereGeometry>(end_R, V, m_bc );
+    
+    //stream according to base class rules
+    ConfinedStreamingMethod<mpcd::detail::SphereGeometry>::stream(timestep);
+
+    //calculating number of particles to evaporate(such that solvent density remain constant)
+    const unsigned int N_remove = round((4.*M_PI/3.)*end_R*end_R*end_R*m_density);
+    const unsigned int N_evap = m_mpcd_pdata->getNGlobal() - N_remove;
```

Comment:
Is it possible that `N_end` will ever be greater than `N_global`? We should guard against this since it is easy:
```suggestion
    const unsigned int N_global = m_mpcd_pdata->getNGlobal();
    const unsigned int N_evap = (N_end < N_global) ? N_global - N_end : 0;
```